BrainPortal/lib/cbrain_task_generators/templates/bourreau.rb.erb

```diff
@@ -494,7 +494,7 @@ class CbrainTask::<%= name %> < <%= (descriptor['custom'] || {})['cbrain:inherit
     # Helper for checking whether outputs are optional
 %   optionalFiles = outputs.select { |o| o['optional'] }
     isOptional = lambda do |f|
-      [<%= optionalFiles.map{ |x| ":'"+x['id']+"'" }.join(",") %>].include?(f)
+      [<%= optionalFiles.map{ |x| ":#{x['id'].to_sym}" }.join(",") %>].include?(f)
```

Comment:
The original code properly generated symbols with single quotes around the symbol name to allow for IDs that contains non-alpha characters, your new code doesn't do that.

Reply:
Not only that, but your code is buggy, it basically does a thing-to-symbol-to-string-to-symbol transformation.